cpp/CMakeLists.txt

```diff
@@ -8,3 +8,4 @@ add_subdirectory(mst)
 add_subdirectory(recognition)
 add_subdirectory(set_cover)
 add_subdirectory(traversal)
+add_subdirectory(cograph_rec)

```

Comment:
A general remark: I was thinking how to place your work so that (1) it shows that it's a common technique, (2) each functionality lands in the proper directory (coloring, independent set, pathwidth, recognition etc.)

Eventually, I think that the best way to do so would be as following:
- create include/techniques/cograph subdirectory (i.e. there will be a general directory "techniques" for some procedures sharing common core, please let me know if you have better naming there)
- there would be an abstract class CographAlgorithm which would build a CoTree,
- CoTree itself would go to include/structures and cpp/structures
- in include/coloring, cpp/coloring and elsewhere there would be a class that inherits from CographAlgorithm and implements a particular run function

If there is no common functionality between algorithms, but just traversing the CoTree, please skip the points 1-2 and go directly to the latter ones.